include/up-cpp/datamodel/builder/Payload.h

```diff
@@ -70,15 +78,22 @@ struct Payload {
 	///          will compile out.
 	/// @param data Data to be serialized and stored.
 	template <typename Serializer, typename ValueT>
-	Payload(Serializer s, const ValueT& data);
+	Payload(Serializer s, const ValueT& data) {
+		auto serializedData = Serializer::serialize(data);
+		//TODO: Payload format needs to be passed as argument or some other way.
+		//TODO: Following line of code is temporary and needs to be removed.
+		auto format = v1::UPayloadFormat::UPAYLOAD_FORMAT_PROTOBUF;
+		payload_ = std::make_tuple(serializedData, format);
```

Comment:
The current serializers in our project, For example UUri serializer, returns std::string and not of type Serialized. My current test case will fail. I want to discuss this usecase with your. Please let me know when you are available for discussion.

Reply:
I must have forgotten to click submit on my previous comment. As discussed, this isn't for use with the current serializers in `datamodel/serializer` - it would require an additional `AsPayloadData` serializer that returns `PayloadBuilder::Serialized`.

A (slightly contrived) example: [compiler explorer](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGIM6SuADJ4DJgAcj4ARpjEIACs8aQADqgKhE4MHt6%2B/ilpGQIhYZEsMXGJtpj2jgJCBEzEBNk%2BfgF2mA6Z9Y0ExRHRsQlJCg1NLbntY32hA2VDiQCUtqhexMjsHOYAzKHI3lgA1CbbbsgIxAKoJ9gmGgCCO3sHmMenBF7J9Dd3j2a7DH2XiOJzco2IoWAPweT0BLzeYIIWFUG2SBGhjwezDYCmSTA2hwAblxjgB2KwPQ5Uw6MHyHfZMBQKQ5yZRMACetFQTHQADESCwmAQyRT7tTxYd7qRfhLqTYZbKWcp7gBNIIAeXuABEAPq89UAJQAsvcACo6g33ADqCqpJlJWpOovtjphDwA9J7DgABKIQzBUQ7KS5EKJeQNeJTM0boEAgcGQw5EQ7ETDJVNKQRJhCvCzsgiYZkEdnJV5US4sX6e92Kw44hRMYCFgB0v0jieUUTzBeZJy1hxjcYTRidvyrXt9/sDYqUELEeAAXph0Ic8ZzuSvGYcmKumHgIUZDqhA1F84XtwwV/xiIKCK2Hu3D0JYnh50uV32B0i4x8vphQZ23aFqQRJcHGrIclyPL8jeQo/NszoPOCXgOIcpp4Gw9wKGy648iKY4aAAnKMQp4MghzPnOtCLsuA4vm%2BmAQIOKAXFccYsHQ1FKGgl5Fos%2BFuoRdyEYBZ7MugQpMKOgnCUwXjJhxtBcZ0bz9neaBeIIECLNJmJCUREkNM2GY0UxNHHhAinKcgiw6QhBH6YRABUECpqEBbEOmmAEDqoijKCWD7MWpaWZxeDcYsTk3BAhlMM2sXabZql1mF3G6cJwmph8xAMF%2BsYgIKADWmA6r%2B9BMd%2BBWoISjGxcsoHgThUF8gKQqNaqGranqhomualpWnZiFCQ6BF6SRjjkcx5yXAwqDsalnQCOg4mFvR1FLhAPGjBRa00eg5gAGwDvx9qirWW3CqJPaHYcsXJcxTboqcGjRQog22hKF0NSAEG4S1sHomYR3Xre92VY9oJcK971unpwl4IG1CtcKYBgJ%2BxKNZBG4wbe7Vqpqur6saZoWtaJ3kg5GVEQQrEAO55XGoSEvO6A6o0wA%2BIwBAQOYZhuIYs3Clgs6vutryOGwhzlqgLCHLNDAALTEEw9Nrs1vMw3DREupTREI4cMWSc26QbSdpwDuZVAVfl01sQVC08ctcapskSWnbrjk05c9PMUzLNs8QHNsFpvNCDRR6BmrG4DoKSmxNmhgpUp4WLbxztphr6Xa9rI2w1ThFyQpDvJS5bmCLEXk%2BX5T1uB7%2BdfVNrGzfNyfcUtCjp8kUXbNghtGQltmax7WVrLljczXN9ut6ny2hdPNlZ8NrqPCN9mw4%2BwBy6whZ4gS49XDq1EeWIChZ/cheoLdqCENpAlitSF9JslXBmJI8QHZGunio/vbbP26GYWwljHkQ5dobX3s3EA6A1ikQEL5RkNcIGTysinR2p8e4QAILZL%2BD95KXwIGYZKADMBYSahuOMwswGMTejgqkI8coMwKkwYqpVPjlQet5UEL0MHMRYNVahtkQIcJrlDHhlU%2BE1SYoIpMZgh4Og4MsWgnB4i8D8NwXgc0OC10sNYAcqx1ivB2DwUgBBNAKOWIVBIGh9CcEkKosxpBNG8A7tY0xHAtDLDgLAGAiAUAy2SHQWI5BKBoBYAE%2BgcRgBSC4HwOgHkO4QCiA4qIoRGjsk4MYlJzBiDsnVFEbQnQ3HGNCcHAg6oGC0HSe43gWAwzAH5kpDu6jSBYEFEYcQ1SWn7kKXgGqTStCBFUJ0eSmxjHuWqA46ifo0keCwA4ggEIWAZN4DVYgUQ0iYC1JgNpwBqJGDMcsKgBhgAKAAGp4EwLTdUpY1HGP4IIEQYh2BSBkIIRQKh1CdN0DEgw%2BzTA6MsPoPAUQO6QGWKgNEmQmkK3VJvBWCsYx9n%2BVYSwXBSQaNWRCYW8BlgdC6M4CArgJhtCSMEWYpRygSFJPkdItQsieFaP4JIqRaWZH6BSoYaKqg1G6NMYlTLuU9LqNMdlgw4hcpIuMBlkwRgivJWKqluL9EbAkIo5R9jOmaMOKoAAHAdBWB1JCHGAMgciUhmwkggLgQgJBjj/C4IsXgbiPEWKsTYjgdjSBLPiDq5s2x/UBsDYGg6pA1EDKcbYEAriDmkC8b41YBBkjyWCZtfxgTiDhG3pwXV%2BrDXGtNYcc1xjlw2qxXoe5whRDiBeRW95agHHfNILTZWyRllqo4Co0NDjNHqnkkm4Ux5tV6oNUak1ZrJAWoNh4MJ6a7XbAdU6g5ywcw8iGNpd1nqlmSC4M2A6e790Hv3dSsNGjODOKjSYpdpBLHxGsUojg2wNXhrPZe6pgj71mCfaejgi633LFWekZwkggA%3D%3D%3D)